src/inventory_ui.cpp

```diff
@@ -2908,3 +2917,79 @@ inventory_selector::stats inventory_drop_selector::get_raw_stats() const
 }
+
+void inventory_examiner::action_examine( const item_location sitem, const bool fit_to_window )
+{
+    if( !fit_to_window ) {
+        inventory_selector::action_examine( sitem );
+    } else {
+        // Code below pulled from the action_examine function in advanced_inv.cpp
+        std::vector<iteminfo> vThisItem;
+        std::vector<iteminfo> vDummy;
+
+        sitem->info( true, vThisItem );
+
+        item_info_data data( sitem->tname(), sitem->type_name(), vThisItem, vDummy, examine_window_scroll );
+        //Tries to create a window covering the right half of the inventory screen
+        data.without_getch = true;
+
+        // NOTE: This assumes that the first column (currently own_inv_column) is used for the item list:
+        int inv_column_width = get_all_columns().front()->get_width();
+        /* NOTE: By flagging force_max_window_size as true, this forces the inventory_selector window to be
+         * TERMX x TERMY , allowing us to assume the window starts at 0,0 and is TERMX by TERMY.  If this is
+         * changed in inventory_selector::prepare_layout, then everything breaks.  This obviously isn't a
+         * great solution */
+
+        int border_width = 1; //Assumed, since the value in inventory_selector is private, but reasonable
+        int width = TERMX - 2 * border_width  - inv_column_width;
+        int height = TERMY  - get_header_height() - 1;
+        point start_position = point( ( inv_column_width + border_width + 1 ),
+                                      get_header_height() + 1 );
+
+        draw_item_info( catacurses::newwin( height, width, start_position ), data );
+    }
+}
+
+item_location inventory_examiner::execute()
+{
+    shared_ptr_fast<ui_adaptor> ui = create_or_get_ui_adaptor();
+    while( true ) {
+        ui_manager::redraw();
+
+        const inventory_entry &selected = get_active_column().get_selected();
+        if( selected ) {
+            const item_location &sitem = selected.any_item();
+            action_examine( sitem, true );
+        }
+
+        const inventory_input input = get_input();
+
+        if( input.entry != nullptr ) {
+            if( select( input.entry->any_item() ) ) {
+                ui_manager::redraw();
+            }
+            return input.entry->any_item();
+        }
+
+        const int scroll_item_info_lines = get_layout_height() - 4;
+
+        if( input.action == "QUIT" ) {
+            return item_location();
+        } else if( input.action == "PAGE_UP" ) {
+            examine_window_scroll -= scroll_item_info_lines;
+        } else if( input.action == "PAGE_DOWN" ) {
+            examine_window_scroll += scroll_item_info_lines;
+        } else if( input.action == "CONFIRM" ) {
+            const inventory_entry &selected = get_active_column().get_selected();
+            if( selected ) {
+                return selected.any_item();
+            }
+        } else if( input.action == "HIDE_CONTENTS" || input.action == "SHOW_CONTENTS" ) {
```

Comment:
You can fold this into the branch below with `else if !(...) { on_input() }`

Reply:
I suppose it could be folded in.  If I understand you correctly, the main result of this would be one fewer check in the loop.  However, the resulting line:
} else if( input.action == "HIDE_CONTENTS" || input.action == "SHOW_CONTENTS" ) { on_input(); }
Feels like it would be harder to parse six months from now.  If that's just my read on it, then I can make the change.

Is there any particular reason why input_context doesn't have a "deregister_action"?  Because that would allow this check to be removed entirely.

Reply:
There probably just wasn't any need for a `deregister_action` so far. 

If you want to disable more actions though, you'll still have to fold them into one of the branches or you'll get a clang-tidy error (duplicate branch).

Reply:
What's the unexpected behavior exactly?

Reply:
Probably the same I found with the pickup menu. Made an issue about that: #51490.

Reply:
This should be fixed in master now; let me know if you still have issues with `HIDE/SHOW_CONTENTS`

Reply:
> This should be fixed in master now; let me know if you still have issues with `HIDE/SHOW_CONTENTS`

Yep.  51727 disables hiding/showing contents in this new screen, so this line is no longer necessary.  I'll add the commit once I've dealt with some new rebasing conflicts.

Reply:
> This should be fixed in master now; let me know if you still have issues with `HIDE/SHOW_CONTENTS`

Looks like I spoke before I'd fully caught up on the changes.  After rebasing, you can use the show/hide contents buttons in this menu, and it won't hide/show/change the items listed, which was the problem before, but it will still _say_ it's hiding things:
![Contents_listed_as_hidden](https://user-images.githubusercontent.com/89038572/140552728-7807d1a9-e4cd-42c8-91ea-590cbd24634c.png)

I've removed the specific block on hiding/showing in my code, because I don't consider this to be a major problem for the menu anymore, but it's still a bit quirky.

Reply:
I can't test atm, but from a quick look I think it's because you're replacing the `ui_adaptor::on_screen_resize()` for the base class so `prepare_layout()` never gets called to hide the entries. You can probably fix that by calling `select_one_of( get_selected().locations)` in your resize callback.

Reply:
https://github.com/CleverRaven/Cataclysm-DDA/pull/52659 will definitely™ fix it. Sorry for the false info in the previous comment.